### ISSUE TITLE
Enforce immutability for incoming task document

### ISSUE THREAD

**Issue by Ayush5120** [elixir-europe/proTES#142]
The incoming task document serves as a log/reference to what the user sent, which is needed for accounting/audit purposes. Therefore, only the parts that are provided by the client should be immutable, and the outgoing request can be modified according to middleware configuration.

**Comment by uniqueg** [comment]
Isn't this already accounted for by #141, @Ayush5120? If so, you can close this issue.

**Comment by Ayush5120** [comment]
No, as discussed this was supposed to be done once the naming issue was fixed and now I can see the naming of the TES task was fixed in #141 , will fix this soon in the upcoming Prs.

**Comment by uniqueg** [comment]
So you really wanna make it immutable. Nice. I like that :)
